tools/doc/allhtml.mjs

```diff
@@ -31,13 +31,31 @@ for (const link of toc.match(/<a.*?>/g)) {
   // Split the doc.
   const match = /(<\/ul>\s*)?<\/\w+>\s*<\w+ id="apicontent">/.exec(data);
 
+  // Get module name
+  const moduleName = href.split('.')[0];
```

Comment:
```suggestion
  const moduleName = href.split('.md', 1)[0];
```

Reply:
Not sure I understand this one. At that point the markdown has already been converted, so all links are referring to `*.html` pages. Did you mean `href.split('.html', 1)[0]`? And if so, why? Is that safer for some reason?

Reply:
Currently we were using `path.basename(href, '.md')`
https://github.com/nodejs/node/blob/21cda8ef92a1038a51bb55d96bd45ec13555aab8/tools/doc/html.mjs#L89

If we introduce a new document whose name would contain a dot (e.g. `package.json.md`), it would cause a problem. Maybe we can do that instead:
```suggestion
  const moduleName = href.replace(/\.html$/, '');
```